DataFormats/SiPixelDigi/interface/PixelErrors.h

```diff
@@ -0,0 +1,21 @@
+#ifndef DataFormats_SiPixelDigi_PixelErrors_h
+#define DataFormats_SiPixelDigi_PixelErrors_h
+
+#include "DataFormats/SiPixelRawData/interface/SiPixelRawDataError.h"
+#include "FWCore/Utilities/interface/typedefs.h"
+
+#include <map>
+#include <vector>
+
+// Better ideas for the placement of these?
+
+struct PixelErrorCompact {
+  uint32_t rawId;
+  uint32_t word;
+  unsigned char errorType;
+  unsigned char fedId;
+};
+
+using PixelFormatterErrors = std::map<cms_uint32_t, std::vector<SiPixelRawDataError>>;
```

Comment:
The idea is to bring this typedef
https://github.com/cms-patatrack/cmssw/blob/ad9b85e79d89f4f6c393c8cc648261366d82adca/EventFilter/SiPixelRawToDigi/interface/PixelDataFormatter.h#L65-L66
into `DataFormats` so that it can be passed between modules via `Event`. It seems to be an intermediate representation of the pixel unpacking errors.

Thinking it again, it seems to only passed around from `SiPixelRawToClusterCUDA` to `SiPixelDigisFromSoA` to be converted to `edm::DetSetVector<SiPixelRawDataError>` there. An alternative would be to produce `edm::DetSetVector<SiPixelRawDataError>` already in `SiPixelRawToClusterCUDA`. I'll try that.

Reply:
> Thinking it again, it seems to only passed around from `SiPixelRawToClusterCUDA` to `SiPixelDigisFromSoA` to be converted to `edm::DetSetVector<SiPixelRawDataError>` there. An alternative would be to produce `edm::DetSetVector<SiPixelRawDataError>` already in `SiPixelRawToClusterCUDA`. I'll try that.

Ok, by quick look going from `std::map<cms_uint32_t, std::vector<SiPixelRawDataError>>` to `edm::DetSetVector<SiPixelRawDataError>` is interleaved with other error data interpretation, so I'd leave those as they are for this PR.

Reply:
The error modeling and storage need to be revised at some point... (TRK DPG?)
we need an example of clients on the GPU side to understand what is required there.